frappe/printing/doctype/network_printer_settings/network_printer_settings.py

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2021, Frappe Technologies and contributors
+# For license information, please see license.txt
+
+import frappe
+from frappe.model.document import Document
+from frappe import _
+
+class NetworkPrinterSettings(Document):
+	@frappe.whitelist()
+	def get_printers_list(self,ip="localhost",port=631):
+		printer_list = []
+		try:
+			import cups
+		except ImportError:
+			frappe.throw(_("You need to install pycups to use this feature!"))
```

Comment:
Please change this message to user friendly message.

And also can you check why cups can not be part of requirements.txt?

Reply:
It's an optional thing. We need to install application software as well. [Doc Link](https://docs.erpnext.com/docs/v12/user/manual/en/setting-up/print/print-settings#3-network-printer-print-server)